### ISSUE TITLE
Doc mask returns negative sparsity

### ISSUE THREAD

**Issue by staghado** [pytorch-labs/attention-gym#93]
The displayed sparsity for the block mask is negative when the block mask is bigger than the max size of the mask.
Similar to this issue #68 

torch nightly : 2.6.0.dev20241221+cu124
repro code :

```
import torch
from torch.nn.attention.flex_attention import create_block_mask

document_id = torch.zeros(100, dtype=torch.int, device="cuda")
document_id[:10] = 0
document_id[10:20] = 1
for i in range(20, 100, 20):
    document_id[i : i + 20] = i // 20 + 1

def document_causal_mask(b, h, q_idx, kv_idx):
    causal_mask = q_idx >= kv_idx
    document_mask = document_id[q_idx] == document_id[kv_idx]
    return causal_mask & document_mask

mask = create_block_mask(document_causal_mask, 1, 1, 100, 100, "cuda")
print(mask)
```

output : 

```
BlockMask(shape=(1, 1, 100, 100), sparsity=-63.84%, 
(0, 0)
██
)
```

does this have an effect on the results obtained with such a mask?

**Comment by drisspg** [comment]
This shud not have any effect on the result. This is an independent code oath from what is used to run w/ flex attention

**Comment by drisspg** [comment]
Can you try upgrading to a newer version of PyTorch we have fixed many bugs since 2.5.1

**Comment by LJHG** [comment]
Hi @NuanBaobao! did you fix this and what is the cause?

**Comment by NuanBaobao** [comment]
> Hi [@NuanBaobao](https://github.com/NuanBaobao)! did you fix this and what is the cause?

@LJHG , I haven't been able to resolve this yet - the issue is more complex than initially expected and I'm still working to identify the underlying cause.

**Comment by drisspg** [comment]
fIxed by ^ pr 

landing now so closing
